query-connector/src/app/database-service.ts

```diff
@@ -453,3 +452,118 @@ export async function insertQuery(input: QueryInput) {
   if (errorArray.length === 0) return { success: true };
   return { success: false, error: errorArray.join(",") };
 }
+
+/**
+ * Function that verifies that a particular value set and all its affiliated
+ * concepts were successfully inserted into the DB. Given a FHIR formatted
+ * value set, the function checks three things:
+ *   1. Whether the value set itself was inserted
+ *   2. Whether each concept included in that value set bundle was inserted
+ *   3. Whether these concepts are now mapped to this value set via foreign key
+ * If any data is found to be missing, it is collected and logged to the user.
+ * @param setToCheck The FHIR formatted value set to check.
+ * @param ersdConcept The eRSD concept this value set belongs to.
+ * @returns A data structure reporting on missing concepts or value set links.
+ */
+export async function checkValueSetInsertion(
+  setToCheck: FhirValueSet,
```

Comment:
Should this be a FhirValueSet or the internal one? For any valuesets that are not from the eRSD, e.g., ones that are manually entered they will be in the internal valueset format (I believe). To make this function extensible to all the valuesets we expect, I think it should expect an internal model of a valueset. 

My understand is that we go: whatever format -> internal vs -> insert so we should always check against the internal model.

Reply:
This makes sense to me. It doesn't impact any of the logic otherwise and keeps us more robust to different input sources so I like this improvement.